flask_monitoringdashboard/database/endpoint.py

```diff
@@ -167,3 +169,28 @@ def get_endpoints_hits(db_session):
         join(Request). \
         group_by(Endpoint.name).\
         order_by(desc(func.count(Request.endpoint_id))).all()
+
+
+def get_avg_duration(db_session, endpoint_id):
```

Comment:
would it make sense to limit this to the last 1K requests? 
i have an endpoint that has 100K requests... and the very
old ones are hardly much relevant... although 1K is also
quite arbitrary... :)

Reply:
This is a very good point and I already addressed it in my thesis (Chapter 4.5). There was virtually no difference in response times between getting the average of 1 entry vs 100k entries.
Of course, this might differ from one rdbms to another, or there might be an arbitrary "breaking point" larger than 100k when this becomes really slow.
I think the best way to prevent this is to add a config setting for the N last requests, with default being all.